frontend/packages/console-shared/src/components/dropdown/ResourceDropdown.tsx

```diff
@@ -140,6 +142,25 @@ class ResourceDropdown extends React.Component<ResourceDropdownProps, State> {
     return true;
   }
 
+  private getResourceList = (nextProps: ResourceDropdownProps) => {
+    const { resources, resourceFilter, dataSelector } = nextProps;
+    const resourceList = {};
+    _.each(resources, ({ data }) => {
+      _.each(data, (resource) => {
+        let key;
+        if (resourceFilter && resourceFilter(resource)) {
+          key = _.get(resource, dataSelector);
+        } else if (!resourceFilter) {
+          key = _.get(resource, dataSelector);
+        }
+        if (key) {
+          resourceList[key] = resource;
```

Comment:
You're squashing duplicate entries of the same name -- which was the explicit issue mentioned in the ticket. Which confuses me even more because you noted that as a comment 🤔  This is an issue we need to fix in addition to adding the badge.

![Screen Shot 2020-09-09 at 12 46 26 PM](https://user-images.githubusercontent.com/8126518/92628314-9d875980-f29a-11ea-9ff8-90c71034d2c9.png)
![Screen Shot 2020-09-09 at 12 46 49 PM](https://user-images.githubusercontent.com/8126518/92628317-9e1ff000-f29a-11ea-8974-c12dff2f53ec.png)

Reply:
In order to avoid the squashing of the entries with same name but different resource, we must need to change the logic of data accumulation in  `getDropdownList` function in ResourceDropdown.  This might require us to change the onChange handlers in some of the ResourceDropdown usages.
Can we have this additional issue addressed in a seperate PR & Ticket ? 
cc: @christianvogt

Reply:
Ah right, the caller defines the firehose and does not actually have the data 🤔

Reply:
So I just added a resource key generator and got it to work:

![image](https://user-images.githubusercontent.com/8126518/92778592-50c18280-f36f-11ea-9309-bf4c3e35ece4.png)

Diff is here if you're curious: https://gist.github.com/andrewballantyne/f0d82506da9aca7ccd688edd0911ac28

Reply:
Now, the generated resource Key will be passed via onChange handler, so the caller has to extract the resourceName from the custom-generated-key or use the new third argument (resource), right?

Reply:
My thought process was ResourceDropdown to implicitly handle multiple fiehose resources , thats why i was talking about a seperate PR, but based on your previous comment, i will update the component with `customResourceKey`  prop.Thanks @andrewballantyne

Reply:
> Now, the generated resource Key will be passed via onChange handler, so the caller has to extract the resourceName from the custom-generated-key or use the new third argument (resource), right?

They crafted it, so yes they will have to extract it... the owner understands the what they did, they can reverse it.